src/main/java/mx/org/corresponsabilidadsocial/api/blog/repositories/PostRepository.java

```diff
@@ -29,4 +32,25 @@ public void setPosts(List<Post> posts) {
         this.posts = posts;
     }
 
+    public boolean existsById(Integer id) {
+
+        for (int i = 0; i < posts.size(); i++) {
+            if (posts.get(i).getId().equals(id)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public Post getPostById(Integer id) {
+        for (int i = 0; i < posts.size(); i++) {
+            if (posts.get(i).getId().equals(id)) {
+                return posts.get(i);
+            }
+        }
+        throw new NotFoundException();
```

Comment:
Podemos agregar un mensaje de error a la excepcion, algo como `String.format("Could not find Post with id: %s", id)`

Reply:
corregido en branch de validaciones